pkg/curatedpackages/regional_registry_test.go

```diff
@@ -2,41 +2,126 @@ package curatedpackages_test
 
 import (
 	"bytes"
+	"context"
 	"io"
 	"net/http"
+	"net/http/httptest"
+	"strings"
 	"testing"
 
 	"github.com/aws/eks-anywhere/pkg/curatedpackages"
 )
 
-func TestTestRegistry(t *testing.T) {
-	err := curatedpackages.TestRegistryWithAuthToken("authToken", "registry_url", func(req *http.Request) (*http.Response, error) {
-		return &http.Response{
-			StatusCode: 200,
-			Body:       io.NopCloser(bytes.NewReader(nil)),
-		}, nil
-	})
-	if err != nil {
-		t.Errorf("Registry is good, but error has been returned %v\n", err)
+func TestTestRegistryWithAuthToken(t *testing.T) {
+	cases := []struct {
+		description string
+		statusCode  int
+		hasError    bool
+	}{
+		{"200 status code does not cause error", 200, false},
+		{"404 status code does not cause error", 404, false},
+		{"400 status code causes error", 400, true},
 	}
 
-	err = curatedpackages.TestRegistryWithAuthToken("authToken", "registry_url", func(req *http.Request) (*http.Response, error) {
-		return &http.Response{
-			StatusCode: 404,
-			Body:       io.NopCloser(bytes.NewReader(nil)),
-		}, nil
-	})
-	if err != nil {
-		t.Errorf("Registry is good, but error has been returned %v\n", err)
+	for _, test := range cases {
+		t.Run(test.description, func(t *testing.T) {
+			err := curatedpackages.TestRegistryWithAuthToken("authToken", "registry_url", func(req *http.Request) (*http.Response, error) {
+				return &http.Response{
+					StatusCode: test.statusCode,
+					Body:       io.NopCloser(bytes.NewReader(nil)),
+				}, nil
+			})
+			if test.hasError && err == nil {
+				t.Errorf("Error should have been returned")
+			}
+			if !test.hasError && err != nil {
+				t.Errorf("Registry is good, but error has been returned %v\n", err)
+			}
+		})
 	}
+}
 
-	err = curatedpackages.TestRegistryWithAuthToken("authToken", "registry_url", func(req *http.Request) (*http.Response, error) {
-		return &http.Response{
-			StatusCode: 400,
-			Body:       io.NopCloser(bytes.NewReader(nil)),
-		}, nil
-	})
-	if err == nil {
-		t.Errorf("Error should have been returned")
+var (
+	goodAwsConfig = "good-aws-config"
+	badAwsConfig  = "bad-aws-config"
+
+	goodAwsKey    = "good-aws-key"
+	goodAwsSecret = "good-aws-secret"
+	badAwsKey     = "bad-aws-key"
+	badAwsSecret  = "bad-aws-secret"
+
+	goodAuthToken = "good-auth-token"
+	badAuthToken  = "bad-auth-token"
+)
+
+type mockRegistryAuthTokenProvider struct{}
+
+func (m *mockRegistryAuthTokenProvider) GetTokenByAWSConfig(ctx context.Context, awsConfig string) (string, error) {
+	if awsConfig == goodAwsConfig {
+		return goodAuthToken, nil
+	}
+	return badAuthToken, nil
+}
+
+func (m *mockRegistryAuthTokenProvider) GetTokenByAWSKeySecret(ctx context.Context, key, secret, region string) (string, error) {
+	if key == goodAwsKey && secret == goodAwsSecret {
+		return goodAuthToken, nil
 	}
+	return badAuthToken, nil
+}
+
+func TestTestAWSConfigRegistryAccessWithAWSConfig(t *testing.T) {
```

Comment:
I don't love the TestTest stutter here, but we've already gone with this as a naming convention so I don't think it's worth refactoring the whole thing. Are these types of Tester objects common in golang? In general it feels funny to me to have Test in a resource name that is not a test.

Reply:
Yeah, the name is funny when you point it out. I don't think it's very common wildly.